aiida/orm/nodes/data/upf.py

```diff
@@ -199,10 +200,13 @@ def parse_upf(fname, check_filename=True):
 
     try:
         upf_contents = fname.read()
-        fname = fname.name
     except AttributeError:
-        with open(fname, encoding='utf8') as handle:
+        with open(fname) as handle:
```

Comment:
I think the encoding should still be specified. Note that the default is locale-dependent - I've been bitten by this more than once :)

Reply:
Is probably the safer choice, except this would still fail if the UPF is not actually utf-8 encoded, and there is no way to specify the correct encoding ....

Reply:
Hmm, what about just exposing it in the `parse_upf` interface (and setting the default to `'utf-8'` there)? Or is that not a user-facing function?

Reply:
Could do, not sure if "officially" it is public facing, but people might be using. On the other hand, I have reimplemented it in `aiida-pseudo` so think to just make sure it is done correctly over there.